marathon_lb.py

```diff
@@ -1555,33 +1555,41 @@ def make_config_valid_and_regenerate(marathon, groups, bind_http_https, ssl_cert
             generated_config = config(apps, groups, bind_http_https, ssl_certs,
                                       templater, haproxy_map, domain_map_array,
                                       app_map_array, config_file)
-            
+
             if not generateAndValidateConfig(generated_config, config_file,
-                                             domain_map_array, app_map_array, haproxy_map):
+                                             domain_map_array, app_map_array,
+                                             haproxy_map):
                 invalid_id = valid_marathon_apps[-1]["id"]
-                logger.warn("invalid configuration caused by app %s; it will be excluded", invalid_id)
+                logger.warn(
+                    "invalid configuration caused by app %s; "
+                    "it will be excluded", invalid_id)
                 excluded_ids.append(invalid_id)
                 del valid_marathon_apps[-1]
 
                 if len(valid_marathon_apps) == 0:
                     apps = []
 
         if len(valid_marathon_apps) > 0:
-            logger.debug("regentrating valid config which excludes the following apps: %s", excluded_ids)
-            compareWriteAndReloadConfig(generated_config, config_file,
-                                        domain_map_array, app_map_array, haproxy_map)
+            logger.debug("regentrating valid config which excludes the"
+                         "following apps: %s", excluded_ids)
+            compareWriteAndReloadConfig(generated_config,
+                                        config_file,
+                                        domain_map_array,
+                                        app_map_array, haproxy_map)
         else:
-            logger.error("A valid config file could not be generated after excluding all apps! skipping reload")
-        
-        logger.debug("regenerating while excluding invalid tasks finished, took %s seconds",
-                        time.time() - start_time)   
+            logger.error("A valid config file could not be generated after"
+                         "excluding all apps! skipping reload")
+
+        logger.debug("regenerating while excluding invalid tasks finished, "
+                     "took %s seconds",
+                     time.time() - start_time)
 
         return apps
 
     except requests.exceptions.ConnectionError as e:
         logger.error("Connection error({0}): {1}".format(
             e.errno, e.strerror))
-    except:
+    except Exception:
```

Comment:
Should we be just logging the exception as a string else won't be able to capture the reason. Same goes for other instances of this exception.

Reply:
The line `logger.exception("Unexpected error!")` right after this one will actually print the exception as well as a stack trace if executed from within an exception context (as this is).

Reply:
Got it !!